[Static Runtime] Fold linear prepack ops

mikeiovine opened the pull request:
Summary: Split `quantized_linear_unpacked_weight_v2` into `linear_prepack` and `quantized_linear` so that the prepacking operation may be eliminated by constant folding.

Test Plan:
Fixes a huge regression in an internal model:

```
Before
        89.6141 ms.    99.0923%. fb::quantized_linear_unpacked_weight_v2 (12 nodes)
After
       0.806852 ms.    53.5365%. quantized::linear (12 nodes, out variant)
(prepacking eliminated)
```

Differential Revision: D39622530

davidberard98 reviewed (approved):
LGTM! also it would be nice to have an OSS test

Commit:
[Static Runtime] Fold linear prepack ops (#85289)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/85289

Split `quantized_linear_unpacked_weight_v2` into `linear_prepack` and `quantized_linear` so that the prepacking operation may be eliminated by constant folding.

Test Plan:
Fixes a huge regression in an internal model:

```
Before
        89.6141 ms.    99.0923%. fb::quantized_linear_unpacked_weight_v2 (12 nodes)
After
       0.806852 ms.    53.5365%. quantized::linear (12 nodes, out variant)
(prepacking eliminated)
```

Reviewed By: tenpercent

Differential Revision: D39622530

fbshipit-source-id: 69f61123ee22763d72ab72f81aa4441bbd8f89c5

mikeiovine commented:
@pytorchbot merge

(Diff landed in Phabricator, not sure why it didn't trigger automatically...)
